esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3721,6 +3722,51 @@ void CWsWuFileHelper::createWUZAPFile(IEspContext& context, IConstWorkUnit* cwu,
         throw MakeStringException(ECLWATCH_CANNOT_COMPRESS_DATA,"Failed to execute system command 'zip'. Please make sure that zip utility is installed.");
 }
 
+void CWsWuFileHelper::createZAPWUQueryAssociatedFiles(IConstWorkUnit* cwu, const char* pathToCreate)
+{
+    Owned<IConstWUQuery> query = cwu->getQuery();
```

Comment:
Any need to check for cwu == nullptr or pathToCreate == nullptr ?

Reply:
The cwu has been checked at the begin of creating ZAP report (CWsWorkunitsSoapBindingEx:: createAndDownloadWUZAPFile). The pathToCreate cannot be nullptr (see my previous comment).